pkg/vm/monitor.go

```diff
@@ -118,6 +125,32 @@ func (m *Module) monitor(ctx context.Context) error {
 	return nil
 }
 
+func (m *Module) cleanupCidata() error {
```

Comment:
This look really good now. I only have one comment:

I think it's important to make sure that this does not race with the creation of VM. Imagine while a machine is still starting (it's cidata has been created but the vm is still not running) the timer ticked and u started the clean up. 

This can cause a big problem, no? Luckily the solution is easy since we already have a lock that is acquired on creation, it's also acquired on `monitor` which also does some clean up.

Just include those 2 lines at the beginning of the function
```go
	m.lock.Lock()
	defer m.lock.Unlock()
```